network/p2p/libp2pConnector.go

```diff
@@ -117,19 +118,32 @@ func (l *libp2pConnector) trimAllConnectionsExcept(peerInfos []peer.AddrInfo) {
 	allCurrentConns := l.host.Network().Conns()
 
 	// for each connection, check if that connection should be trimmed
+ConnectionLoop:
 	for _, conn := range allCurrentConns {
 
 		// get the remote peer ID for this connection
 		peerID := conn.RemotePeer()
 
 		// check if the peer ID is included in the current fanout
 		if peersToKeep[peerID] {
-			continue
+			continue ConnectionLoop
 		}
 
 		peerInfo := l.host.Network().Peerstore().PeerInfo(peerID)
 		log := l.log.With().Str("remote_peer", peerInfo.String()).Logger()
 
+		// retain the connection if there is a Flow One-to-One stream on that connection
+		// (we do not want to sever a connection with on going direct one-to-one traffic)
+		for _, s := range conn.GetStreams() {
```

Comment:
We could also encapsulate this for loop in a func that returns a bool, maybe a function `HasOneToOneConnection`, which would eliminate the need for `ConnectionLoop` label.

Might be a bit cleaner, but overall same functionality.